rpcs3/rpcs3qt/gui_application.cpp

```diff
@@ -161,19 +161,16 @@ bool gui_application::Init()
 	{
 		welcome_dialog* welcome = new welcome_dialog(m_gui_settings, false);
 
-		bool use_dark_theme = false;
-
-		connect(welcome, &QDialog::accepted, this, [&]()
+		connect(welcome, &QDialog::finished, this, [&]()
```

Comment:
Why the change to 'finished'? if the user rejects our terms then we should terminate execution instead.

Reply:
the signals in this `connect` and the `connect` in `welcome_dialog.cpp` needed to be swapped. Both are executed but the sequence is `accepted` (when you press on the `Continue` button) and then `terminated` (where the settings on the dialog are finally used). The bug was simply due to wrong signals registration.

Reply:
I see, I was specifically referring to the scenera in which the user closes the dialog forcefully, without agreeing to "I have read the Quickstart guide".
Can you add this code in this pr for the first time RPCS3 shows up?

connect(welcome, &QDialog::rejected, this, []() { std::exit(0); });

std::exit should be safe to use here.

Reply:
no pronlem to add that `connect`. But is that scenario related to the case where the rpcs3 app is forced to terminate when the welcome dialog is active? Current welcome dialog does not allow to close the dialog without accepting the agreement (the `continue` button is disabled)

Reply:
At this point you can just remove this connect all together and just move the guisettings call into the connect in the dialog

Reply:
yes, especially when other gui settings (e.g. `m_gui_settings->SetValue(gui::ib_show_welcome, QVariant(!checked));`) are managed directly on the dialog. Maybe, in the past, the logic was quite different needing a more articulated code.
Anyway, as last step from my side on this PR, just moving the logic on the dialog maintaining the compliancy to the current logic.